---
title: "Assign taxonomy to the plastids subset for Bacterial 16S amplicons"
output: html_notebook
---



### Overview: 
1. Plastid sequences were identified as 'Chloroplast' at the Order level in the Silva138-assigned taxonomy for the 202008 MarineB data set
2. This subset of plastid sequences were exported as a rank abundance table with columns (ASV, ASVid, chlT)  chlT=chloroplasts sum count
3. The taxonomic assignments to phytoplankton lineages was refined using the `assignTaxonomy` function from Dada2 using the PR2.12.0 16S database (PhytoRef) with default parameters + `outputBoostraps`. *No bootstrap filtering was applied* so pay attention to that when observing the results.
4. the plastid taxonomy was merged with the plastid Rank Abundance Table and then merged directly with the working version of the bacterial 16S data for the NRS (20080814)

See below for resulting colnames (this is pre-loaded data)

```{r}
colnames(pelBnrsPlast)

colnames(pelEpm)

plastids <- left_join(plastids, plastids.tax.df)

write_csv(plastids, '~/plastids_rank_abundance_table_20201101.csv')


```
# select core columns on latest long ASV table (90% Collapsed)
(i.e. remove GTDB taxa columns) and join with Silva 138 Reference taxonomy

calculate the totals for each sample (abundT (all seqs), chlT (all Chloroplasts), cyaT (all cyanobacteriia [incl ChlT]))

mutate these totals onto the long format df

check the sanity and get users to double check for duplications

```{r}
pelB <- read_csv('/Volumes/Dropbox/MarineMicrobes Dropbox/amplicons2020/MarineBASVlongGTDB.20200914v3.csv',  guess_max=100000)

pelBref <- read_csv('/Volumes/Dropbox/MarineMicrobes Dropbox/amplicons2020/MarineBrefASV_Silva138_20200708.csv', guess_max=100000)

colnames(pelBtest)

pelBtest <- pelB %>% select(-c(contains('tax')))
pelBtest <- pelBtest %>% select(-c(contains('boot')))


pelB<- left_join(pelBtest, pelBref)


totals<- pelB %>% group_by(code) %>% summarise(abundT=sum(abund))

Chl.totals<- pelB %>%  filter(tax.Order=='Chloroplast') %>% group_by(code) %>% summarise(chlT=sum(abund))

Cya.totals<- pelB %>%  filter(tax.Class=='Cyanobacteriia') %>% group_by(code) %>% summarise(cyaT=sum(abund))

pelB <- left_join(pelB, totals)
pelB <- left_join(pelB, Chl.totals)
pelB <- left_join(pelB, Cya.totals)

table(pelB$tax.Order)

plastids<- read_csv('/Volumes/Dropbox/MarineMicrobes Dropbox/amplicons2020/MarineBChlASVRef.20200914v3.csv', guess_max=100000)

plastids.tax <- assignTaxonomy(plastids$ASV, refFasta='/Volumes/Dropbox/MarineMicrobes\ Dropbox/pr2_version_4.12.0_16S_dada2.fasta', outputBootstraps = T, taxLevels = c("Kingdom", "Supergroup","Division", "Class", "Order", "Family", "Genus", "Species"), tryRC = T)

plastids.tax.df <- as.data.frame(plastids.tax)

plastids.tax.df$ASV <- rownames(plastids.tax)

pelBnrs<- pelB %>% filter(uniqcode %in% nrs)

pelBnrsPlast <- pelBnrs %>% filter(ASV %in% plastids.tax.df$ASV)

pelEpm <- left_join(pelBnrsPlast, plastids.tax.df, by='ASV')

colnames(plastids.tax.df)

```
write data for the full Bacterial ASV table with Silva138 taxonomic lineages


write data for the subset of plastid ASV table with PR2 16S taxonomic lineages



```{r}


meta$code <- as.character(meta$code)

meta$utc_date_sampled_yyyymmdd <- as.Date(meta$utc_date_sampled_yyyymmdd, '%d/%m/%y')
pelBmeta<- pelB %>% left_join(meta) %>% arrange(uniqcode, utc_date_sampled_yyyymmdd, depth_m)

pelBG <- pelBmeta %>% group_by(tax.Genus, code) %>% summarise(divG=sum(abund)) 
pelBG <- pelBG %>% spread(key=tax.Genus, value=divG, fill=0)
pelBGm <- pelBG %>% gather (key = 'tax.Genus', value='divG', -code)
pelBGm<- pelBGm %>% left_join(meta) %>% arrange(uniqcode, utc_date_sampled_yyyymmdd, depth_m)

pelBmetaAll <- pelBmeta %>% select(-c(asvT.orig, asvT.orig.y, asvT.orig.x)) %>% distinct()
 

pelBmetaChl <- pelBmetaAll %>% filter(ASV %in% plastids.tax.df$ASV) %>%  select(-c(contains('tax'))) %>% select(-c(contains('boot'))) %>% left_join(plastids.tax.df)

write_csv(pelBmetaAll, '~/MarineB_all_20201103.csv')

write_csv(pelBmetaChl, '~/MarineB_Chl_20201103.csv')
 
```



```{r, fig.width=12, fig.height=12}
q <-ggplot(pelBGm %>% 
         filter(uniqcode %in% c("MAI", "PHB", "ROT", "NSI", "YON"), depth_m< 5, year > 2014), aes(x=utc_date_sampled_yyyymmdd,  
                 y=divG, fill=fct_lump(tax.Genus, 80, w=divG, other_level = NA))) + 
  geom_area( 
             stat='identity', position='fill')   +
facet_grid(uniqcode ~ .) +theme_bw() +   coord_polar() +
  scale_fill_manual(values=mycol1000[seq(990, 20,-6)], name='Top 80 Genera', ) + 
  scale_size_area(max_size = 10, name='Relative Abundance', breaks=c(0,100,500,1000, 10000, 50000))  + 
  labs(y='RA', title='Top 80  Genera', x='Date Sampled') + theme( strip.text.y = element_text(angle = 0), panel.spacing.x = unit(0.01,"cm"), panel.spacing.y = unit(0.2,"cm")) + geom_point(aes(y=1.05))



q
```

```{r, fig.width=12, fig.height=12}
pelBC <- pelBmeta %>% group_by(tax.Class, code) %>% summarise(divC=sum(abund)) 
pelBC <- pelBC %>% spread(key=tax.Class, value=divC, fill=0)
pelBCm <- pelBC %>% gather (key = 'tax.Class', value='divC', -code)
pelBCm<- pelBCm %>% left_join(meta) %>% arrange(uniqcode, utc_date_sampled_yyyymmdd, depth_m)
```


```{r, fig.width=12, fig.height=12}
c <-ggplot(pelBCm %>% 
         filter(uniqcode %in% c("MAI", "PHB", "ROT", "NSI", "YON"), depth_m ==50, year > 2014), aes(x=utc_date_sampled_yyyymmdd,  
                 y=divC, fill=fct_lump(tax.Class, 80, w=divC, other_level = NA)), color='black', lwd=0.1) + 
  geom_area( 
             stat='identity', position='fill')   +
facet_grid(uniqcode ~ .) +theme_bw() +   coord_polar() +
  scale_fill_manual(values=mycol1000[seq(400, 990,6)], name='Top 80 Classes', ) + 
  scale_size_area(max_size = 10, name='Relative Abundance', breaks=c(0,100,500,1000, 10000, 50000))  + 
  labs(y='RA', title='Top 80  Classes', x='Date Sampled') + theme( strip.text.y = element_text(angle = 0), panel.spacing.x = unit(0.01,"cm"), panel.spacing.y = unit(0.2,"cm")) + geom_point(aes(y=1.05))



#summary(pelBCm$day_length)

c

ggsave('~/MarineB_NRS_by_class_D.pdf', height=20, width=20)
```
```{r, fig.width=12, fig.height=12}
pelBAm<- pelB %>% filter(uniqcode %in%  c("MAI", "PHB", "ROT", "NSI", "YON")) %>% left_join(meta) %>% arrange(uniqcode, utc_date_sampled_yyyymmdd, depth_m)

ggplot(pelBAm %>% 
         filter( !is.na(ASVid),uniqcode %in% c("MAI", "PHB", "ROT", "NSI", "YON"), depth_m < 5, year > 2014), aes(x=utc_date_sampled_yyyymmdd,  
                 y=(abund/abundT)*100, fill=fct_lump(ASVid, 100, w=(abund/abundT), other_level = NA))) + 
  geom_area( 
             stat='identity', position='fill')   +
facet_grid(uniqcode ~ .) +theme_bw() +   coord_polar() +
  scale_fill_manual(values=mycol1000[seq(1000, 1,-2)], name='Top 40 ASVs', ) + 
  #scale_size_area(max_size = 10, name='Relative Abundance', breaks=c(0,100,500,1000, 10000, 50000))  + 
  labs(y='RA', title='Class', x='Date Sampled') + theme( strip.text.y = element_text(angle = 0), panel.spacing.x = unit(0.01,"cm"), panel.spacing.y = unit(0.2,"cm")) + geom_point(aes(y=1.05),shape=24, size=0.2)  



geom_point(aes(y=day_length/20), color='red',size=0.2) 

geom_path(data=pelBCm %>% 
         filter(uniqcode %in% c("MAI", "PHB", "ROT", "NSI", "YON"), depth_m< 5, year > 2014) %>% distinct(utc_date_sampled_yyyymmdd, .keep_all=T), aes(y=temperature_deg_c/30), color='white',size=1, alpha=0.4)


#summary(pelBCm$day_length)

a

pelBAm %>% 
         filter(!is.na(ASVid),uniqcode %in% c("MAI", "PHB", "ROT", "NSI", "YON"), depth_m< 5, year > 2014)

table(pelBAm$tax.Family)
```

